### DISCUSSION TITLE
Does ArtifactHub support index.yaml with OCI references?

### DISCUSSION THREAD

**Discussion by carrodher** [artifacthub/hub#4055] [Q&A]
Hi,

We're exploring whether ArtifactHub supports an _index.yaml_ where the chart references are in OCI format, as shown below:
```diff
    urls:
-    - https://charts.bitnami.com/bitnami/airflow-18.3.2.tgz
+    - oci://registry-1.docker.io/bitnamicharts/airflow:18.3.2
```

We are currently testing this change and have encountered no issues when using the Helm client (version 3.7.0 or later). At this stage, we’re verifying compatibility with other services that consume the Bitnami _index.yaml_.

Would you be able to confirm if ArtifactHub supports this format?

Thank you!

**Comment by tegioz** [comment]
Hi @carrodher 👋

Yes, that should be possible.

Out of curiosity, I've checked if someone else had published a repository using this approach and [there is one](https://artifacthub.io/packages/search?repo=bougoucharts). The repository was added as an HTTP based repo, using this url <https://bougou.github.io/charts/>. But if you inspect [its index you'll see some OCI references](https://bougou.github.io/charts/index.yaml).

In any case, before you update it in Artifact Hub, we'd be happy to test it ourselves in a separate Artifact Hub instance to verify everything is all right before you do the switch in production. You can share with us the repository url when you have it ready.

Another point to keep in mind is any possible rate limit that the OCI registry you use may impose. Bitnami's charts repo is quite large, and during its processing we send multiple requests to the source concurrently. So it'd be great if you can make sure this won't be a problem either.

**Comment by carrodher** [comment]
Sorry @tegioz, we couldn’t provide a testing _index.yaml_ earlier. We’ve been making the changes gradually and have now started releasing new Helm charts in full OCI format, updating the _index.yaml_ to use OCI references instead of HTTPS.

For example, a new WordPress version has been released and included in the _index.yaml_ as follows:
```yaml
  wordpress:
    ...
    urls:
    - oci://registry-1.docker.io/bitnamicharts/wordpress:24.0.5
    version: 24.0.5
```

I’ve double-checked, and everything looks fine in ArtifactHub:
* The repository is being properly ingested:
> Last processed: 3 hours ago(next check in ~ 30 minutes)
> Last security scan: 15 minutes ago(next scan in ~ 15 minutes)
* The new WordPress version (24.0.5) is available at [ArtifactHub](https://artifacthub.io/packages/helm/bitnami/wordpress) as usual.

This confirms that replacing HTTPS references with OCI is a smooth and functional transition.

**Reply by tegioz** [reply]
Awesome, thanks for the update! 👍
